msprime/mutations.py

```diff
@@ -27,71 +27,82 @@
 from . import utils
 from . import provenance
 
-# TEMPORARY
 import numpy as np
 
-# Alphabets for mutations.
-BINARY = 0
-NUCLEOTIDES = 1
+
+class MutationModel(_msprime.MutationModel):
+    """
+    Superclass of mutation models. Allows you to build your own mutation model.
+
+    TODO document properly.
+    """
+    def asdict(self):
+        # This version of asdict makes sure that we have sufficient parameters
+        # to call the contructor and recreate the class. However, this means
+        # that subclasses *must* have an instance variable of the same name.
+        return {
+            key: getattr(self, key) for key in inspect.signature(
+                self.__init__).parameters.keys()
+        }
 
 
-# FIXME Dunno what the right API is here yet, just putting this in to get stuff
-# working.
-def mutation_model_factory(alphabet=BINARY):
-    if alphabet == BINARY:
+class Binary(MutationModel):
```

Comment:
maybe `BinaryAlleles`, since just `msprime.Binary` is not very descriptive?

Reply:
Good call.

Reply:
I've changed to ``UnidirectionalBinary``, as ``BinaryAlleles`` doesn't quite capture it. What do you think?

Reply:
It's not unidirectional - 1s change to 0s (otherwise subsequent mutations would be invisible!); it's just that the ancestral state is always labeled `0`. I think `BinaryAlleles` is fine as long as we document this fact well.